staging/src/k8s.io/apiserver/pkg/apis/audit/v1alpha1/types.go

```diff
@@ -0,0 +1,163 @@
+	// RequestURI is the request URI as sent by the client to a server.
+	RequestURI string `json:"requestURI" protobuf:"bytes,5,opt,name=requestURI"`
+	// Verb is the kubernetes verb associated with the request.
+	// For non-resource requests, this is identical to HttpMethod.
+	Verb string `json:"verb" protobuf:"bytes,6,opt,name=verb"`
+	// Authenticated user information.
+	User authnv1.UserInfo `json:"user" protobuf:"bytes,7,opt,name=user"`
+	// Impersonated user information.
+	// +optional
+	Impersonate *authnv1.UserInfo `json:"impersonate,omitempty" protobuf:"bytes,8,opt,name=impersonate"`
+	// Source IP, from where the request originates.
+	// +optional
+	SourceIP string `json:"sourceIP,omitempty" protobuf:"bytes,9,opt,name=sourceIP"`
+	// Object reference this request is targeted at.
+	// Does not apply for List-type requests, or non-resource requests.
+	// +optional
+	ObjectRef *apiv1.ObjectReference `json:"objectRef,omitempty" protobuf:"bytes,10,opt,name=objectRef"`
+	// The response status, populated even when the ResponseObject is not a Status type.
+	// For successful responses, this will only include the Code and StatusSuccess.
+	// For non-status type error responses, this will be auto-populated with the error Message.
+	// +optional
+	ResponseStatus *metav1.Status `json:"responseStatus,omitempty" protobuf:"bytes,11,opt,name=responseStatus"`
+
+	// API object from the request, in JSON format. The RequestObject is recorded as-is in the request
+	// (possibly re-encoded as JSON), prior to version conversion, defaulting, admission or
+	// merging. It is an external versioned object type, and may not be a valid object on its own.
+	// Omitted for non-resource requests.  Only logged at RequestObject Level and higher.
+	// +optional
+	RequestBody string `json:"requestBody,omitempty" protobuf:"bytes,12,opt,name=requestBody"`
+	// API object returned in the response, in JSON. The ResponseObject is recorded after conversion
+	// to the external type, and serialized as JSON.  Omitted for non-resource requests.  Only logged
+	// at ResponseObject Level and higher.
+	// +optional
+	ResponseBody string `json:"responseBody,omitempty" protobuf:"bytes,13,opt,name=responseBody"`
+}
+
+// EventList is a list of audit Events.
+type EventList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Items []Event `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// Policy defines the configuration of audit logging, and the rules for how different request
+// categories are logged.
+type Policy struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Rules specify the audit Level a request should be recorded at.
+	// A request may match multiple rules, in which case the FIRST matching rule is used.
+	// The default audit level is None, but can be overridden by a catch-all rule at the end of the list.
+	Rules []PolicyRule `json:"rules" protobuf:"bytes,2,rep,name=rules"`
```

Comment:
I'm worried that making Policy a list type makes the type to rigid (not future proof). Also I don't think PolicyRule conceptually makes sense as a standalone type, especially since they're strictly ordered in a policy. I added a `PolicyList` instead.